oauthproxy.go

```diff
@@ -92,7 +94,12 @@ func (u *UpstreamProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		r.Header.Set("GAP-Auth", w.Header().Get("GAP-Auth"))
 		u.auth.SignRequest(r)
 	}
-	u.handler.ServeHTTP(w, r)
+	if u.wsHandler != nil && r.Header.Get("Upgrade") == "websocket" {
```

Comment:
We might want to check for the Connection header being set to "upgrade" as well.

Reply:
Good idea.